From 1ab845e8c7b48162fb6d3bf1b3e7fd0fca8654de Mon Sep 17 00:00:00 2001
From: uros <uros@138bc75d-0d04-0410-961f-82ee72b054a4>
Date: Mon, 2 Jun 2014 22:05:48 +0000
Subject: [PATCH 024/134] 	Backport from mainline 	2014-06-02  Uros
 Bizjak  <ubizjak@gmail.com>

	PR target/61239
	* config/i386/i386.c (ix86_expand_vec_perm) [case V32QImode]: Use
	GEN_INT (-128) instead of GEN_INT (128) to set MSB of QImode constant.



git-svn-id: svn+ssh://gcc.gnu.org/svn/gcc/branches/gcc-4_8-branch@211150 138bc75d-0d04-0410-961f-82ee72b054a4
---
 gcc/ChangeLog          | 9 +++++++++
 gcc/config/i386/i386.c | 2 +-
 2 files changed, 10 insertions(+), 1 deletion(-)

diff --git a/gcc/ChangeLog b/gcc/ChangeLog
index 8d9a352..d2b0352 100644
--- a/gcc/ChangeLog
+++ b/gcc/ChangeLog
@@ -1,3 +1,12 @@
+2014-06-03  Uros Bizjak  <ubizjak@gmail.com>
+
+	Backport from mainline
+	2014-06-02  Uros Bizjak  <ubizjak@gmail.com>
+
+	PR target/61239
+	* config/i386/i386.c (ix86_expand_vec_perm) [case V32QImode]: Use
+	GEN_INT (-128) instead of GEN_INT (128) to set MSB of QImode constant.
+
 2014-05-28  Guozhi Wei  <carrot@google.com>
 
 	PR target/61202
diff --git a/gcc/config/i386/i386.c b/gcc/config/i386/i386.c
index 8ca9a03..fa81d0f 100644
--- a/gcc/config/i386/i386.c
+++ b/gcc/config/i386/i386.c
@@ -20505,7 +20505,7 @@ ix86_expand_vec_perm (rtx operands[])
 	  t1 = gen_reg_rtx (V32QImode);
 	  t2 = gen_reg_rtx (V32QImode);
 	  t3 = gen_reg_rtx (V32QImode);
-	  vt2 = GEN_INT (128);
+	  vt2 = GEN_INT (-128);
 	  for (i = 0; i < 32; i++)
 	    vec[i] = vt2;
 	  vt = gen_rtx_CONST_VECTOR (V32QImode, gen_rtvec_v (32, vec));
-- 
1.8.5.5

